integration/Earthfile

```
@@ -1,35 +1,66 @@
-FROM hseeberger/scala-sbt:11.0.6_1.3.10_2.13.1
+FROM docker:19.03.7-dind
 WORKDIR /scala-example
+RUN apk add openjdk11 curl docker-compose bash
+
+#Scala
+# Defaults if not specified in --build-arg
+ARG sbt_version=1.3.2
+ARG sbt_home=/usr/local/sbt
+
+# Download and extract from archive
+RUN apk add --no-cache --update wget
+RUN mkdir -pv "$sbt_home"
+RUN wget -qO - "https://github.com/sbt/sbt/releases/download/v$sbt_version/sbt-$sbt_version.tgz" >/tmp/sbt.tgz
+RUN tar xzf /tmp/sbt.tgz -C "$sbt_home" --strip-components=1
+RUN ln -sv "$sbt_home"/bin/sbt /usr/bin/
+
+# This triggers a bunch of useful downloads.
+RUN sbt sbtVersion
 
 deps:
     COPY build.sbt ./
     COPY project project
-    RUN sbt update
+    #run sbt for caching purposes
+    RUN touch a.scala && sbt compile && rm a.scala
     SAVE IMAGE
 
 build:
     FROM +deps
     COPY src src
     RUN sbt compile
+    SAVE IMAGE
 
-test:
-    FROM +deps
+unit-test:
+    FROM +build
     COPY src src
     RUN sbt test
 
-integration_test:
-    FROM +deps
+integration-test:
+    FROM +build
     COPY src src
-    COPY docker-compose.yml ./
-     WITH DOCKER 
-        RUN docker-compose up -d ;\
-            sbt test && \
+    COPY docker-compose.yml ./ 
+    WITH DOCKER 
+        RUN docker-compose up -d && \
+            while ! nc -z localhost 5432; do sleep 1; done; \
```

maybe we can change it to wait at most 10 seconds, to avoid an infinite loop?
```suggestion
            for i in {1..10}; do nc -z localhost 5432 && break; sleep 1; done
```